
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Methylotrophic Origin of Life &#8212; Methylotrophy Origin of Life</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=3ee479438cf8b5e0d341" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=3ee479438cf8b5e0d341" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=3ee479438cf8b5e0d341" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=3ee479438cf8b5e0d341"></script>

    <script src="_static/documentation_options.js?v=125c9c8b"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'index';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Load and subset the dataframe of COGs" href="Notebooks/1_COG_CountRefSpecies.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="#">
  
  
  
  
  
  
    <p class="title logo__title">Methylotrophy Origin of Life</p>
  
</a></div>
        <div class="sidebar-primary-item"><ul class="navbar-icon-links navbar-nav"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/Philipgdv/Methylotrophy" title="GitHub" class="nav-link" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><span><i class="fa-brands fa-square-github fa-lg" aria-hidden="true"></i></span>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn navbar-btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1 current active">
                <a class="reference internal" href="#">
                    Methylotrophic Origin of Life
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Scripts:</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html">Load and subset the dataframe of COGs</a></li>

<li class="toctree-l1"><a class="reference internal" href="Notebooks/2_COG_PresenceTaxa.html">Plot the NCBI classification tree from file</a></li>


<li class="toctree-l1"><a class="reference internal" href="Notebooks/3_COG_CoOccurence.html">Load the OG DataFrame</a></li>


<li class="toctree-l1"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html">Manually add the archaeal COGs, use root level of EggNOG 6</a></li>


</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/Philipgdv/Methylotrophy" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/Philipgdv/Methylotrophy/edit/master/index.md" target="_blank"
   class="btn btn-sm btn-source-edit-button dropdown-item"
   title="Suggest edit"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-pencil-alt"></i>
  </span>
<span class="btn__text-container">Suggest edit</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/index.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Methylotrophic Origin of Life</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Methylotrophic Origin of Life</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#system-requirements">System Requirements</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#hardware-requirements">Hardware requirements</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#software-requirements">Software requirements</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#os-requirements">OS Requirements</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#command-line-tools">Command line tools</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#python-dependencies">Python Dependencies</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#installation-guide">Installation Guide:</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#instructions">Instructions:</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-on-the-demo-data">Run on the demo data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-the-full-analysis">Run the full analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#initiate">1 - Initiate</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#snakefile">2 - Snakefile</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#mapping">3 - Mapping</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#combining-counts">4 - Combining Counts</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#analyses">5 - Analyses</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#license">License</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#scripts">Scripts</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="methylotrophic-origin-of-life">
<h1>Methylotrophic Origin of Life<a class="headerlink" href="#methylotrophic-origin-of-life" title="Link to this heading">#</a></h1>
<p>This repository contains the scripts and notebooks used in used in the manuscript:</p>
<p>Methylotrophic Acetyl-CoA Formation as the Pathway at the Origin of Life,
By Philip J. Gorter de Vries, Alex Toftgaard Nielsen, Alfred M. Spormann</p>
<ul class="simple">
<li><p><a class="reference internal" href="#system-requirements"><span class="xref myst">System Requirements</span></a></p></li>
<li><p><a class="reference internal" href="#installation-guide"><span class="xref myst">Installation Guide</span></a></p></li>
<li><p><a class="reference internal" href="#instructions"><span class="xref myst">Instructions</span></a></p></li>
<li><p><a class="reference external" href="https://matplotlib.org/stable/project/license.html#license" rel="noopener noreferrer" target="_blank" title="Matplotlib 3.9.0"><span class="xref myst">License</span></a></p></li>
</ul>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="system-requirements">
<h1>System Requirements<a class="headerlink" href="#system-requirements" title="Link to this heading">#</a></h1>
<section id="hardware-requirements">
<h2>Hardware requirements<a class="headerlink" href="#hardware-requirements" title="Link to this heading">#</a></h2>
<p>The code requires only a standard computer with enough RAM to support the in-memory operations. The analysis was run locally on a laptop computer with a 1,4 GHz Quad-Core Intel Core i5 running MacOS Sonoma 14.4.1</p>
</section>
<section id="software-requirements">
<h2>Software requirements<a class="headerlink" href="#software-requirements" title="Link to this heading">#</a></h2>
<section id="os-requirements">
<h3>OS Requirements<a class="headerlink" href="#os-requirements" title="Link to this heading">#</a></h3>
<p>This code is supported for <em>macOS</em>, it has been tested on the following systems: macOS: Sonoma (14.4.1)</p>
</section>
<section id="command-line-tools">
<h3>Command line tools<a class="headerlink" href="#command-line-tools" title="Link to this heading">#</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span> <span class="mf">7.18.2</span>
<span class="n">python</span> <span class="mf">3.9.15</span>
<span class="n">Orthofinder</span> <span class="mf">2.5.4</span>
<span class="n">MAFFT</span> <span class="n">v7</span><span class="mf">.508</span>
<span class="n">fasttree</span> <span class="mf">2.1.11</span>
</pre></div>
</div>
</section>
<section id="python-dependencies">
<h3>Python Dependencies<a class="headerlink" href="#python-dependencies" title="Link to this heading">#</a></h3>
<p>The included notebooks mainly depend on the Python packages:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">numpy</span> <span class="mf">1.23.5</span> 
<span class="n">pandas</span> <span class="mf">1.5.2</span> 
<span class="n">scipy</span> <span class="mf">1.10.1</span>
<span class="n">scikit</span><span class="o">-</span><span class="n">learn</span> <span class="mf">1.2.0</span> 
<span class="n">fastcluster</span> <span class="mf">1.2.6</span>
<span class="n">ete3</span> <span class="mf">3.1.2</span>
</pre></div>
</div>
</section>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="installation-guide">
<h1>Installation Guide:<a class="headerlink" href="#installation-guide" title="Link to this heading">#</a></h1>
<p>All packages can be installed with conda by running in the shell the command (replacing <em>PackageName</em>):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">*</span><span class="n">PackageName</span><span class="o">*</span>
</pre></div>
</div>
<p>To run the code in the jupyter notebooks, run (replacing <em>EnvironmentName</em>):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">anaconda</span> <span class="n">ipykernel</span>
<span class="n">python</span> <span class="o">-</span><span class="n">m</span> <span class="n">ipykernel</span> <span class="n">install</span> <span class="o">--</span><span class="n">user</span> <span class="o">--</span><span class="n">name</span><span class="o">=</span> <span class="o">*</span><span class="n">EnvironmentName</span><span class="o">*</span>
</pre></div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="instructions">
<h1>Instructions:<a class="headerlink" href="#instructions" title="Link to this heading">#</a></h1>
<p>The repository consists of a series of scripts that will combine user-selected genomes with genomes of the EggNOG database to create a single dataframe of gene counts per species per orthologous gene group. This dataframe is used for various analyses and visualizations in the jupyter notebooks. The graphical abstract shows how these notebooks are used in the manuscript.</p>
<p><img alt="alt text" src="https://github.com/Philipgdv/Methylotrophy/blob/main/Figures/GraphicalAbstract.png?raw=true" /></p>
<section id="run-on-the-demo-data">
<h2>Run on the demo data<a class="headerlink" href="#run-on-the-demo-data" title="Link to this heading">#</a></h2>
<p>The repository is filled with demo data, allowing to test the notebooks. The demo file is a subset of the full dataframe The notebooks are:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span><span class="n">_COG_CountRefSpecies</span><span class="o">.</span><span class="n">ipynb</span>
<span class="mi">2</span><span class="n">_COG_PresenceTaxa</span><span class="o">.</span><span class="n">ipynb</span>
<span class="mi">3</span><span class="n">_COG_Clustering</span><span class="o">.</span><span class="n">ipynb</span>
<span class="mi">4</span><span class="n">_LGT_MSA</span><span class="o">&amp;</span><span class="n">Tree</span><span class="o">.</span><span class="n">ipynb</span>
</pre></div>
</div>
</section>
<section id="run-the-full-analysis">
<h2>Run the full analysis<a class="headerlink" href="#run-the-full-analysis" title="Link to this heading">#</a></h2>
<p>If you wish to run the full analysis and recreate the full dataframe, run the following steps:</p>
<section id="initiate">
<h3>1 - Initiate<a class="headerlink" href="#initiate" title="Link to this heading">#</a></h3>
<p>Change directory to the Methylotrophy folder &amp; initiate it by running in the terminal (Where 1935183 is the taxon number of Asgardarchaea):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">/</span><span class="n">Methylotrophy</span>
<span class="n">python</span> <span class="n">Scripts</span><span class="o">/</span><span class="n">Setup</span><span class="o">.</span><span class="n">py</span> <span class="mi">1935183</span>
</pre></div>
</div>
</section>
<section id="snakefile">
<h3>2 - Snakefile<a class="headerlink" href="#snakefile" title="Link to this heading">#</a></h3>
<p>run the snakefile to run OrthoFinder</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span> <span class="o">-</span><span class="n">j</span> <span class="mi">4</span>
</pre></div>
</div>
</section>
<section id="mapping">
<h3>3 - Mapping<a class="headerlink" href="#mapping" title="Link to this heading">#</a></h3>
<p>the workflow will produce a FASTA file containing the sequences of one of each identified orthology group find the file at:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">/</span><span class="n">Methylotrophy</span><span class="o">/</span><span class="n">Data</span><span class="o">/</span><span class="n">OFResults</span><span class="o">/</span><span class="n">MultipleSequenceAlignments</span><span class="o">/</span><span class="n">OFOG_Representatives</span><span class="o">.</span><span class="n">fasta</span> 	
</pre></div>
</div>
<p>Upload this file to http://eggnog-mapper.embl.de/ to link the found OrthoFinder groups to COG groups. Download the output as a csv or tsv file and move it to /Methylotrophy/Data/input/</p>
</section>
<section id="combining-counts">
<h3>4 - Combining Counts<a class="headerlink" href="#combining-counts" title="Link to this heading">#</a></h3>
<p>Run a python script to download all COGs from EggNOG at root level and link the identified OrthoFinder orthologous groups to them, creating a count matrix</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">Scripts</span><span class="o">/</span><span class="n">COG_counts</span><span class="o">.</span><span class="n">py</span>
</pre></div>
</div>
</section>
<section id="analyses">
<h3>5 - Analyses<a class="headerlink" href="#analyses" title="Link to this heading">#</a></h3>
<p>Run the analyses in the notebooks, as described in the materials &amp; methods section.</p>
</section>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="license">
<h1>License<a class="headerlink" href="#license" title="Link to this heading">#</a></h1>
<p>This project is covered under the <strong>Apache 2.0 License</strong>.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="scripts">
<h1>Scripts<a class="headerlink" href="#scripts" title="Link to this heading">#</a></h1>
<div class="toctree-wrapper compound">
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Scripts:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html">Load and subset the dataframe of COGs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html#list-of-wlp-related-cogs">list of WLP-related COGs</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html#subset-to-only-the-reference-species-and-wlp-genes">Subset to only the reference species and WLP genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html#find-species-names">Find species names</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html#build-the-tree-with-feature-annotation">Build the tree with feature annotation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/1_COG_CountRefSpecies.html#plot-separately-for-the-figure">Plot Separately for the figure</a></li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/2_COG_PresenceTaxa.html">Plot the NCBI classification tree from file</a></li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/2_COG_PresenceTaxa.html#load-the-og-dataframe">Load the OG DataFrame</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/2_COG_PresenceTaxa.html#open-the-list-of-clusters-of-orthologous-groups-of-interest">Open the list of clusters of orthologous groups of interest</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/2_COG_PresenceTaxa.html#by-genus">By Genus</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/2_COG_PresenceTaxa.html#group-counts-per-order-and-convert-to-percentage-presence">Group counts per order and convert to percentage presence</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/3_COG_CoOccurence.html">Load the OG DataFrame</a></li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/3_COG_CoOccurence.html#compute-the-jaccard-pairwise-distance">Compute the Jaccard pairwise distance</a></li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/3_COG_CoOccurence.html#create-hierarchichal-clusters">Create Hierarchichal Clusters</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/3_COG_CoOccurence.html#plot-the-largest-cluster">Plot the largest cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/3_COG_CoOccurence.html#plot-a-cluster-of-choice">Plot a cluster of choice</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html">Manually add the archaeal COGs, use root level of EggNOG 6</a></li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#download-and-gather-sequences-align-and-build-tree">Download and gather sequences, align and build tree</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#download-the-bacterial-level-msa-from-eggnog">Download the bacterial level MSA from EggNOG</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#download-the-corresponding-archaeal-level-msas-from-eggnog">Download the corresponding Archaeal level MSAs from EggNOG</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#copy-the-orthofinder-groups-and-add-taxonomic-id-for-asgard">Copy the OrthoFinder groups and add Taxonomic ID for Asgard</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#concatenate-all-sequences-to-add-into-a-single-fasta">Concatenate all sequences to add into a single FASTA</a></li>
<li class="toctree-l2"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#combine-in-a-single-msa-using-mafft">Combine in a single MSA using MAFFT</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Notebooks/4_LGT_MSA%26Tree.html#load-the-tree-and-color-it-by-phylogeny">Load the tree and color it by phylogeny</a></li>
</ul>
</div>
</section>


                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="right-next"
       href="Notebooks/1_COG_CountRefSpecies.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Load and subset the dataframe of COGs</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Methylotrophic Origin of Life</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#system-requirements">System Requirements</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#hardware-requirements">Hardware requirements</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#software-requirements">Software requirements</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#os-requirements">OS Requirements</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#command-line-tools">Command line tools</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#python-dependencies">Python Dependencies</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#installation-guide">Installation Guide:</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#instructions">Instructions:</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-on-the-demo-data">Run on the demo data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-the-full-analysis">Run the full analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#initiate">1 - Initiate</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#snakefile">2 - Snakefile</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#mapping">3 - Mapping</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#combining-counts">4 - Combining Counts</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#analyses">5 - Analyses</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#license">License</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#scripts">Scripts</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Philip Gorter de Vries
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024, Philip Gorter de Vries.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=3ee479438cf8b5e0d341"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=3ee479438cf8b5e0d341"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>